### PR TITLE
Update bs-platform & Move to @rescript/react

### PR DESCRIPTION
**Pull request by jeong-sik**
image

Related.
https://github.com/reasonml-labs/bs-css/issues/236

All test passed

### CODE REVIEW

**Comment by giraud** [comment]
This one is problematic. it will prevent devs that still use reasonml to upgrade, and actually it prevents devs using rescript to upgrade. 
I don't know what the best solution is

**Comment by giraud** [comment]
I think there is a need to have new modules for rescript-react.
for ex, `bs-css-emotion` could point to `rescript-react` and have a new package `bs-css-emotion-reason` that still use `reason-react` for compatibility. It would break `package.json` but fortunately upgrade would be extra simple.

File: bs-css-dom/package.json [hunk 1]
```diff
@@ -23,11 +23,11 @@
...
   "peerDependencies": {
-    "react": "^17.0.0",
-    "react-dom": "^17.0.0",
-    "reason-react": "^0.9.0"
+    "@rescript/react": "^0.10.2",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2"
   },
   "devDependencies": {
-    "bs-platform": "^8.0.0"
```
Comment (Cfarrell111) [inline bs-css-dom/package.json:31]: -

**Review by Cfarrell111 — APPROVED** [review]

**Comment by giraud** [comment]
updated master with rescript

**Pull request closed without merging**